File: src/main/java/com/blogly/blogly/domain/user/Email.java
package com.blogly.blogly.domain.user;

import java.util.regex.Pattern;

public record Email(String value) {

    private static final Pattern EMAIL_PATTERN = Pattern.compile("[A-Za-z0-9+_.-]+@(.+)");

    public Email {
        if (value == null || !EMAIL_PATTERN.matcher(value).matches()) {
            throw new InvalidEmailException(value);
        }
    }
}

File: src/main/java/com/blogly/blogly/domain/user/Email.kt
package com.blogly.blogly.domain.user

@JvmInline
value class Email(val value: String) {
    init {
        require(EMAIL_PATTERN.matches(value)) { "Invalid email format: $value" }
    }

    companion object {
        private val EMAIL_PATTERN = Regex("[A-Za-z0-9+_.-]+@(.+)")
    }
}
